core/src/main/java/org/bitcoinj/store/SPVBlockStore.java

```diff
@@ -162,6 +162,7 @@ else if (fileLength < randomAccessFile.length())
             // Check or initialize the header bytes to ensure we don't try to open some random file.
             if (exists) {
                 byte[] currentHeader = new byte[4];
+                ((Buffer) buffer).rewind();
```

Comment:
I don't think calling `rewind` is necessary, because `channel.map()` has just been called and `buffer` was just initialized.

(It also is not necessary in the not-exists case, but since `initNewStore` is called elsewhere the rewind is needed in `initNewStore`)

Reply:
It's meant to be defensive, as in my other PR I'm moving stuff around and suddenly this rewind() became essential.

Reply:
It's ok to leave it there, but `buffer` is initialized just 5 lines above.